pkg/core/region.go

```diff
@@ -838,6 +862,8 @@ func (r *RegionsInfo) CheckAndPutRegion(region *RegionInfo) []*RegionInfo {
 	origin, overlaps, rangeChanged := r.setRegionLocked(region, true, ols...)
 	r.t.Unlock()
 	r.UpdateSubTree(region, origin, overlaps, rangeChanged)
+	// FromStorage means this region's meta info might be stale.
+	r.AtomicAddStaleRegionCnt()
```

Comment:
Here we need to make sure that only reload from storage will use `CheckAndPutRegion`.

Reply:
fixed by return regions num